docs/proposals/14-advanced-scheduling.md

```diff
@@ -0,0 +1,77 @@
+
+* `seed.gardener.cloud/not-ready`: The seed is `NotReady`. 
+* `seed.gardener.cloud/over-capacity`: The seed's capacity for shoots has been exceeded, see [Ensuring Seeds Capacity for Shoots Is Not Exceeded](13-automated-seed-management.md#ensuring-seeds-capacity-for-shoots-is-not-exceeded).
+
+### Evicting Shoots from Drained Seeds
+
+The next step in migrating shoots away from an unhealthy seed is actually draining the seed, which involves evicting all shoots from this seed. This could be achieved in one of the following ways:
+
+* By manually marking all seed shoots as orphaned, either by adding an `orphan` annotation or simply be deleting the `seedName` field in the shoot specification.
+* By a special `gardenctl` command, for example `gardenctl drain <seed>`, that would automate the action described above. It should also add the `seed.gardener.cloud/unschedulable` taint to the specified seed.
+
+**Note:** The behavior described here is similar to the behavior of the `kubectl drain <node>` command. This command adds a taint with a `NoSchedule` effect to the node, but doesn't add any taints with a `NoExecute` effect. Instead, it simply forcibly evicts all pods from the node.
+
+### Scheduling Orphaned Shoots to Healthy Seeds 
+
+The next step in migrating shoots away from an unhealthy seed is rescheduling all orphaned shoots onto healthy seeds by `gardener-scheduler`. The scheduler should ensure, among other things, that:
+
+* A shoot with the `orphan` annotation or with an empty `seedName` field is rescheduled.
+* A shoot is never scheduled onto a seed tainted with a taint with a `NoSchedule` effect (unless the shoot tolerates the taint).
+* The new seed capacity for shoots is not exceeded by adding a new shoot, see [Ensuring Seeds Capacity for Shoots Is Not Exceeded](13-automated-seed-management.md#ensuring-seeds-capacity-for-shoots-is-not-exceeded). This could perhaps also be achieved by automatically tainting the seed with a `NoSchedule` taint once its capacity is exceeded by the `gardenlet`.
+
+To reschedule a shoot, the scheduler updates the `seedName` field in the shoot specification with the name of the new seed. 
+
+### Migrating the Shoots to the New Seeds
+
+The final step is actually migrating the shoots control planes from the old to the new seeds. The migration is triggered by the updates to the `seedName` field in the shoot specification by the scheduler described above.
+
+The migration process is described in more detail in [GEP-7: Shoot Control Plane Migration](07-shoot-control-plane-migration.md). 
+
+## Descheduling Shoots from Healthy but Overloaded Seeds
+
+In certain situations, for example if the seed capacity for shoots drops below its initial value for whatever reason, it might be required to deschedule shoots from an otherwise healthy but already overloaded seed. In this case, only a certain number of shoots should be descheduled from the overloaded seed and scheduled onto other seeds. 
```

Comment:
Hmm, I don't think, that there is some foreseeable need to automate a "partial Seed drain". Once the others sections are implemented, the operator is able to do this manually, if such a rare situation arises, where it is needed.

Reply:
Should I remove the entire section then? Or do you suggest to rework it in some way?

Reply:
Keeping the section for now, until further feedback. It's anyway just a mention of something that could be valuable, without going into any kind of detail.

Reply:
Yes, fine for me :)